target node 6.10.0 and upgrade benv and some other dependencies

jasoncalabrese opened the pull request.

jasoncalabrese commented:
@shanselman @MilosKozak @amazaheri I'm trying to get the NS dependencies updated and for the first pass I'm updating the node version in package.json from 0.10.x to 4.6.x along with some of the dependencies.

It was tested on heroku without issue, but @ELUTE saw deploy failures on azure.

If you have any ideas or could give it a try that would be great

ELUTE commented:
![screen shot 2016-11-17 at 4 48 44 pm](https://cloud.githubusercontent.com/assets/7684327/20410880/2801fd88-aced-11e6-946f-dab5edfa6e89.png)
This is the error I get. Site is created but fails to deploy.

shanselman commented:
What's the log say? That error isn't enough info.

On Thu, Nov 17, 2016 at 2:42 PM, Kate Farnsworth <notifications@github.com>
wrote:

> [image: screen shot 2016-11-17 at 4 48 44 pm]
> <https://cloud.githubusercontent.com/assets/7684327/20410880/2801fd88-aced-11e6-946f-dab5edfa6e89.png>
> This is the error I get. Site is created but fails to deploy.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/2071#issuecomment-261393287>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAALTEY5qhLTPFaFqOQtmUG16C8RnmQXks5q_NhQgaJpZM4KUbHy>
> .
>



-- 
Scott Hanselman
Donate to fight diabetes: http://hnsl.mn/fightdiabetes

ELUTE commented:
Let me know which logs you are looking for and I can run them for you.

shanselman commented:
You can usually get to the logs by doing the deploy with Application Logs
turned on in Azure:
https://blogs.msdn.microsoft.com/benjaminperkins/2014/04/01/get-azure-website-deployment-logs-from-kudu/


On Fri, Nov 18, 2016 at 12:32 PM, Kate Farnsworth <notifications@github.com>
wrote:

> Let me know which logs you are looking for and I can run them for you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/2071#issuecomment-261634638>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAALTHFRVhzV9kSPxX4RfvsuGdfrFgNUks5q_gtigaJpZM4KUbHy>
> .
>



-- 
Scott Hanselman
Donate to fight diabetes: http://hnsl.mn/fightdiabetes

LoriLori commented:
Did you try to run on Free azure? Might run out of free CPU quota in 2 minutes :(

PieterGit commented:
@jasoncalabrese @shanselman any news about when/if this will be merged to dev/master? 
nodejs v0.10 and v0.12 are now officially end-of-life. nodejs v4 will be maintained until 2018-04-01
(source https://en.wikipedia.org/wiki/Node.js#Releases )

Will nodejs v6 also be supported for Nightscout?

Did anybody tried installing node v4 on Azure recently?

shanselman commented:
Azure doesn't need Node v4 installed. Azure has all these Nodes already installed (see below) and Nightscout would need to tell it which one in package.json:

```
"engines": {
    "node": "6.9.1"
}, 
```

This is the list of Node runtimes on Azure:

```
{
"nodejs": [
{
"version": "0.10.18",
"npm": "1.3.8"
},
{
"version": "0.10.21",
"npm": "1.3.11"
},
{
"version": "0.10.24",
"npm": "1.3.21"
},
{
"version": "0.10.26",
"npm": "1.4.3"
},
{
"version": "0.10.28",
"npm": "1.4.9"
},
{
"version": "0.10.29",
"npm": "1.4.10"
},
{
"version": "0.10.31",
"npm": "1.4.23"
},
{
"version": "0.10.32",
"npm": "1.4.28"
},
{
"version": "0.10.40",
"npm": "1.4.28"
},
{
"version": "0.10.5",
"npm": "1.2.18"
},
{
"version": "0.12.0",
"npm": "2.5.1"
},
{
"version": "0.12.2",
"npm": "2.7.4"
},
{
"version": "0.12.3",
"npm": "2.9.1"
},
{
"version": "0.12.6",
"npm": "2.11.2"
},
{
"version": "0.6.20",
"npm": "1.1.37"
},
{
"version": "0.8.19",
"npm": "1.2.30"
},
{
"version": "0.8.2",
"npm": "1.1.36"
},
{
"version": "0.8.26",
"npm": "1.2.30"
},
{
"version": "0.8.27",
"npm": "1.2.30"
},
{
"version": "0.8.28",
"npm": "1.2.30"
},
{
"version": "4.0.0",
"npm": "2.14.2"
},
{
"version": "4.1.0",
"npm": "2.14.3"
},
{
"version": "4.1.2",
"npm": "2.14.4"
},
{
"version": "4.2.1",
"npm": "3.3.9"
},
{
"version": "4.2.2",
"npm": "3.3.12"
},
{
"version": "4.2.3",
"npm": "3.5.1"
},
{
"version": "4.2.4",
"npm": "2.14.12"
},
{
"version": "4.3.0",
"npm": "2.14.12"
},
{
"version": "4.3.2",
"npm": "2.14.12"
},
{
"version": "4.4.0",
"npm": "2.14.20"
},
{
"version": "4.4.1",
"npm": "2.14.20"
},
{
"version": "4.4.6",
"npm": "2.15.5"
},
{
"version": "4.4.7",
"npm": "2.15.8"
},
{
"version": "4.5.0",
"npm": "2.15.9"
},
{
"version": "4.6.0",
"npm": "2.15.9"
},
{
"version": "4.6.1",
"npm": "2.15.9"
},
{
"version": "5.0.0",
"npm": "3.3.9"
},
{
"version": "5.1.1",
"npm": "3.3.12"
},
{
"version": "5.3.0",
"npm": "3.3.12"
},
{
"version": "5.4.0",
"npm": "3.3.12"
},
{
"version": "5.5.0",
"npm": "3.3.12"
},
{
"version": "5.6.0",
"npm": "3.6.0"
},
{
"version": "5.7.0",
"npm": "3.6.0"
},
{
"version": "5.7.1",
"npm": "3.6.0"
},
{
"version": "5.8.0",
"npm": "3.7.3"
},
{
"version": "5.9.1",
"npm": "3.7.3"
},
{
"version": "6.0.0",
"npm": "3.8.6"
},
{
"version": "6.1.0",
"npm": "3.8.6"
},
{
"version": "6.10.0",
"npm": "3.10.10"
},
{
"version": "6.2.2",
"npm": "3.9.5"
},
{
"version": "6.3.0",
"npm": "3.10.3"
},
{
"version": "6.5.0",
"npm": "3.10.3"
},
{
"version": "6.6.0",
"npm": "3.10.3"
},
{
"version": "6.7.0",
"npm": "3.10.3"
},
{
"version": "6.9.0",
"npm": "3.10.8"
},
{
"version": "6.9.1",
"npm": "3.10.8"
},
{
"version": "6.9.2",
"npm": "3.10.9"
},
{
"version": "6.9.4",
"npm": "3.10.10"
},
{
"version": "6.9.5",
"npm": "3.10.10"
},
{
"version": "7.0.0",
"npm": "3.10.8"
},
{
"version": "7.1.0",
"npm": "3.10.9"
},
{
"version": "7.2.0",
"npm": "3.10.9"
},
{
"version": "7.3.0",
"npm": "3.10.10"
},
{
"version": "7.4.0",
"npm": "4.0.5"
},
{
"version": "7.5.0",
"npm": "4.1.2"
},
{
"version": "7.6.0",
"npm": "4.1.2"
}
]
}
```

sulkaharo commented:
Scott where are you pulling the supported node list from? I tried to find a document that lists this but couldn't find it.

shanselman commented:
@sulkaharo from https://YOURSITE.scm.azurewebsites.net/api/diagnostics/runtime

jasoncalabrese commented:
Maybe we should go directly to 6.10.0

jasoncalabrese commented:
I tested 6.10.0 locally and seems to work good, I don't have a good way to test on Heroku or Azure now.

sulkaharo commented:
I have a Heroku node I can use for testing, checking

jasoncalabrese commented:
Think I fixed a bug with posting an array of treatments thats been making the tests flaky for a long time.  I think there have been other report of upsert/dedupe not working

jasoncalabrese commented:
@shanselman I haven't used azure and don't understand any of the azure specific scripts/config that we have, but it sounds like you're saying that me might not need that script anymore.  I'd really like to get this merged to dev.  If you could make the changes to get the deploy to work or recruit someone else to help it would be great.  I've been ignoring NS for several months, but if we can get this wrapped up I'd like to continue upgrading the other dependencies.

PieterGit commented:
I installed https://github.com/nightscout/cgm-remote-monitor/pull/2071 on my local system.
I noticed I have an somewhat older, but not ancient node/npm version. Do I need to use 6.10.0 LTS version, or should it also work with node 4.5.0?
I didn't notice any problems with the `wip/upgrade` branch, only a couple of warnings during install.
It installs fine, and I didn't notice any errors with node 4.5.0 (yet).

```
$ npm install --production
npm WARN engine share2nightscout-bridge@0.1.4: wanted: {"node":"~0.10.0"} (current: {"node":"4.5.0","npm":"2.15.9"})
npm WARN engine minimed-connect-to-nightscout@1.1.0: wanted: {"node":"0.10.x"} (current: {"node":"4.5.0","npm":"2.15.9"})
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
...
npm WARN optional dep failed, continuing fsevents@1.1.1
npm WARN deprecated buffer-browserify@0.1.0: Package not maintained. Recent browserify uses https://github.com/feross/buffer
...

I'll try later with 6.10.0 LTS on a new system, but I was curious if it's meant to be that it seems to work fine with node 4.5.0 as well... I don't want to upgrade node on my production site, but first check if it works without problems.

sulkaharo commented:
Tried running locally and on Heroku and not having any issues. Browserify feels slower than before, although amazingly it ran faster in Heroku than locally (which should never happen; not sure why this is the case).

PieterGit commented:
Also did the test part on 4.5.0. 3 tests failed:
```
$ npm --version
2.15.9
$ node --version
v4.5.0
$ npm test
...
# tests 282
# pass 279
# fail 3
Makefile:44: recipe for target 'test' failed
make: *** [test] Error 3
npm ERR! Test failed.  See above for more details.
$ grep "not ok" test.txt
not ok 66 client open careportal, and enter a treatment
not ok 204 Profile editor should produce some html
not ok 224 reports should produce some html
```

jasoncalabrese commented:
I just tried the tests on node 4.5.0 after `rm -rf node_modules` and `npm install` and they passed

shanselman commented:
@jasoncalabrese OK I'll try to take a look Monday or Tuesday. I think Ali might be able to look also. Not sure of his GitHub @

jasoncalabrese commented:
I'm running `wip/upgrade` in production with node 6.1.0

jasoncalabrese commented:
@shanselman maybe just removing the `deploy.sh` would help? or is there a generic deploy.sh that azure needs?  It doesn't seem like we should need anything special if we're using a supported node version

sulkaharo commented:
Read up a little: shouldn't we go directly to the LTS release of the 6 series (Boron)?

jasoncalabrese commented:
I meant to write 6.10.0, current LTS, I'm also using nvm locally and on my droplet so it will be easy to change the version

shanselman commented:
Trying this today...sorry for the delay. I've been having NS run at 100% CPU and been having issues.

jasoncalabrese commented:
Thanks @shanselman, I've seen NS peg the cpu some too, in my case removing a tuning option (--max_old_space_size=300) that I was testing seemed to help.  I'm hoping that getting to a current version node will help with performance some, same for the other dependencies.

shanselman commented:
Where do I remove that from? Not sure if I'm using it but it's 100% all the time now.

shanselman commented:
@jasoncalabrese Any chance we could Skype this afternoon/evening? My node ability is limited but my Azure isn't. Between us we could bang this out in an hour.

jasoncalabrese commented:
I was setting in my init script on my digitalocean droplet, don't know if it's possible with azure or even heroku.

This evening should work for me, I'm hoping that the current setup with the `deploy.sh` script is just really out of date and there's a new way where we don't need that script at all.  Would be great if azure only needed `package.json`

shanselman commented:
@jasoncalabrese Yes, that deploy.sh is years old and likely not needed. I just don't know how to do the Git branchy/deployy stuff. I'm Pacific. When is a good time for you?

bewest commented:
I wrote the build script. At the time it was created it was needed to
ensure we get the deployed versioning info as well as running bower.
On Mon, Mar 27, 2017 at 12:02 PM Scott Hanselman <notifications@github.com>
wrote:

> @jasoncalabrese <https://github.com/jasoncalabrese> Any chance we could
> Skype this afternoon/evening? My node ability is limited but my Azure
> isn't. Between us we could bang this out in an hour.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/2071#issuecomment-289544993>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYDww_0lXFSRQun0XU-IckubDWY1bvgks5rqAIHgaJpZM4KUbHy>
> .
>

stavlor commented:
I just switched to wip/upgrade will let you know if i see any issues:
crm@nightscout:~/cgm-remote-monitor$ node --version
v6.10.1
crm@nightscout:~/cgm-remote-monitor$ npm --version
3.10.10

jasoncalabrese commented:
@shanselman I'm PDT too, anytime between 4:30 and 7 works for me, maybe @bewest could join us :)

I did upgrade bower as part of this, but that was a little while ago and there's probably something newer that would work too

shanselman commented:
@bewest Cool. That deployment script you there is auto-generated by Kudu in Azure. It's also generated by azure-cli with `azure site deploymentscript --node -t bash`

Looks like you've added this line? 

`echo "\"$SCM_COMMIT_ID\"" > $DEPLOYMENT_SOURCE/scm-commit-id.json
`

As an FYI, @jasoncalabrese Azure will make the deploy.sh (exactly as it's checked in, MINUS that added line) automatically. It's the addition of that that to get the commitid that makes the whole file end up checked in. 

I'm asking the Kudu team if we can get rid of that line.

jasoncalabrese commented:
That's used here: https://github.com/nightscout/cgm-remote-monitor/blob/master/env.js#L82, maybe there's another env var or some other way to get the commit id?  I'm not sure if heroku shows the right commit id

shanselman commented:
@jasoncalabrese Can you check your NS install and look in the lower right corner, all the way at the bottom. It's down there.

jasoncalabrese commented:
My NS site is running on a linux box at digitalocean, so it has no problem getting the commit id.  I think first we should get NS to deploy to azure with a current version of node.

After that we can see if we can get the correct commit id to display, but since deploys are usually from a fork the commit id isn't really that useful since a merge will add a new commit id.

bewest commented:
Thanks Scott. That line was added (after pairing with you) specifically to
provide the runtime environment with versioning info so that our help
screen could correctly identify and trace the exact version of software
running.
On Mon, Mar 27, 2017 at 12:54 PM Jason Calabrese <notifications@github.com>
wrote:

> I was setting in my init script on my digitalocean droplet, don't know if
> it's possible with azure or even heroku.
>
> This evening should work for me, I'm hoping that the current setup with
> the deploy.sh script is just really out of date and there's a new way
> where we don't need that script at all. Would be great if azure only needed
> package.json
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/2071#issuecomment-289546533>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYDw-SYA4sRcUiJaGnYVMpYFmD27vT0ks5rqAM_gaJpZM4KUbHy>
> .
>

bewest commented:
If kudu can provide the runtime environment with all the correct startup
tasks as well as the version info, the entire script can be removed
trivially.

Sounds like last issue is the exact version info e.g. From `git rev-parse
HEAD`.

On Mon, Mar 27, 2017 at 12:58 PM Benjamin West <bewest@gmail.com> wrote:

> Thanks Scott. That line was added (after pairing with you) specifically to
> provide the runtime environment with versioning info so that our help
> screen could correctly identify and trace the exact version of software
> running.
> On Mon, Mar 27, 2017 at 12:54 PM Jason Calabrese <notifications@github.com>
> wrote:
>
> I was setting in my init script on my digitalocean droplet, don't know if
> it's possible with azure or even heroku.
>
> This evening should work for me, I'm hoping that the current setup with
> the deploy.sh script is just really out of date and there's a new way
> where we don't need that script at all. Would be great if azure only needed
> package.json
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/2071#issuecomment-289546533>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYDw-SYA4sRcUiJaGnYVMpYFmD27vT0ks5rqAM_gaJpZM4KUbHy>
> .
>
>

shanselman commented:
@jasoncalabrese Cool. I'm deploying from a local wip/upgrade now to Azure. Will report back.

@bewest Totally hear you and I remember. Things change in a year and it'd be nice if we could handle both requirements without the custom file. Looking at this now.

jasoncalabrese commented:
@ELUTE, @shanselman was able to get this to deploy to his site, it might be that the version of node I had tried first wasn't supported by Azure back when you last tried, but seems that the currently selected version will work.  If you or anyone else can see if this deploys now it would be great.

Also there's a setting in Azure `WEBSITE_NODE_DEFAULT_VERSION` that @shanselman removed since it would override the version we select in `package.json`

bewest commented:
Should also be in the status.json endpoint.
On Mon, Mar 27, 2017 at 1:57 PM Jason Calabrese <notifications@github.com>
wrote:

> That's used here:
> https://github.com/nightscout/cgm-remote-monitor/blob/master/env.js#L82,
> maybe there's another env var or some other way to get the commit id? I'm
> not sure if heroku shows the right commit id
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nightscout/cgm-remote-monitor/pull/2071#issuecomment-289562103>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYDw37cq1OevGR4n2Zjm6oUEycT3vuBks5rqBBCgaJpZM4KUbHy>
> .
>

shanselman commented:
UPDATE: Yes, I was totally able to update to this wip/upgrade version nicely just by pushing from here, rather than dev (where I was before). I DID have to remove `WEBSITE_NODE_DEFAULT_VERSION` which, oddly, was set to 0.10. No idea when this was set or who set it. I removed it. 

When I deployed, I went over to Azure and looked at the deployment logs and it shows it correctly selecting the newer node.exe. 

![image](https://cloud.githubusercontent.com/assets/2892/24427608/7dff68bc-13c0-11e7-8c9a-270313956135.png)

I confirmed this at runtime by visiting https://YOURSITEscm.azurewebsites.net/ProcessExplorer/ and looking at the properties of node.exe.

The CPU and memory chart looks much nicer and the site (anecdotally) is at least 2x faster for me.

![image](https://cloud.githubusercontent.com/assets/2892/24427690/bf5e46fc-13c0-11e7-8bb9-393c84895ca5.png)

shanselman commented:
FYI this is working for me on Azure and I am able (or, at least, I have once) taken a heapsnapshot successfully. I'll do another in a few hours and compare.

stavlor commented:
Good to know I've encountered no issues so far with this branch. All working well on node 6.10.1.

jasoncalabrese commented:
@stavlor where are you running NS, your own server?

If anyone would like to try looking at heap dumps, you can now add `dumps` to your `ENABLE`, and then hit /api/v2/dumps/start and a file will be created, can can then download and open it in chrome dev tools

stavlor commented:
Yeah I'm running on a gcloud vm

shanselman commented:
OK, I've got two heapsnapshots, one 33 megs and one 6 hours later that is 54 megs!

I'm not sure how to read this (email me and I'll send them to any of you. Not sure if there is PII in them) but there are tons of "maps" and arrays that are new but aren't getting collected:

![image](https://cloud.githubusercontent.com/assets/2892/24442411/d63d046a-1412-11e7-8527-5d58f7f8b3da.png)

jasoncalabrese commented:
I've been looking at some of my snapshots too, it was seeing a big increase in compiled code, but need to compare different intervals.  My last snapshot is 105M, and I expect it to keep going.  I'm going to look at it some more later.

jasoncalabrese commented:
This is what I'm seeing when comparing heap dumps, not sure what to make of it yet, but System Objects are growing

![heaps](https://cloud.githubusercontent.com/assets/751143/24484602/e33b733a-14b3-11e7-9874-550cfdb2b0ad.gif)

shanselman commented:
Do we know anyone on the nodejs V8 team? I did some profiler stuff over a timeperiod and I see this:

![image](https://cloud.githubusercontent.com/assets/2892/24883911/e8faa272-1dfb-11e7-98da-c1cd5495881b.png)

roblourens commented:
If you want to take a CPU profile of the JS code, you can use https://github.com/node-inspector/v8-profiler

It won't show much if the CPU usage is just from GC or V8 but you can check whether there's something in JS doing more work than it should be.

shanselman commented:
FYI, for those who are watching this epic saga. I skyped with @roblourens today and he said that something in the heapsnapshot pointed to a BSON parser. I renamed all my Mongo Collections to .bak and started over and my site's CPU has DROPPED.

**Working Theory:** Something is bad in one of my collections and when that piece of data gets loaded/hit something in NS goes nuts. I'm going to rename each collection back one at a time and see if I can find the bad actor.

![image](https://cloud.githubusercontent.com/assets/2892/24892704/7444fe5e-1e33-11e7-9c7c-4101b9cf6ff3.png)

Question to @jasoncalabrese and others. Where should I be looking? profile? or deep in one of the larger collections?

MilosKozak commented:
Mostly the problem is treatment collection

sulkaharo commented:
Less data is guaranteee to cause the CPU load to drop. One big reason is the periodic Mongo polling, which reloads the two days of data frequently to see if there's been inserts into the past, which then gets compared to in-memory data to get the delta to send to clients. Some treatment types are loaded for up to three weeks to get things like the sensor age. If you see the CPU load to stay low after running on the reduced dataset for >2 days, that'll be interesting. Alternatively, if you can pinpoint that most of the load comes from one of device statuses, sgv data of treatments, that'd be valuable info.

sulkaharo commented:
Jason or Milos, any idea if the REST API will gracefully handle data updates if we create a simple experimental build which disables the Mongo polls? Scott do you use rest for all data updates or are you injecting data directly to Mongo?

shanselman commented:
@MilosKozak Can you expand on your comment? There's not enough info to understand what you mean?

@sulkaharo I already have multiple weeks of date showing it gets slower and slower linearly over about 5 days until it's useless. I propose that most Nightscout+Loop users are not seeing this because they are either on Heroku and rebooting each day or they are on Azure shared instances that are also restarting. Here's five days. You can see when I deleted stuff on the far right and that it's starting again:
![image](https://cloud.githubusercontent.com/assets/2892/24930391/2f1a3dae-1ebe-11e7-9a6a-edce6a09b878.png)

Are you thinking, @sulkaharo if I removed the *AGE plugins then this could go away? This is what I have now.

`bridge loop pump careportal iob cage basal sage iage cob alexa dumps`

It'll be great to figure this CPU/Memory leak out so NS will get better for everyone!

jasoncalabrese commented:
When I was comparing heap dumps it looked like strings were staying even, but system objects were growing.  I'm not sure what would cause that

shanselman commented:
@sulkaharo OK, it was 3-4 months of data. I renamed all the collections so I'm now running it with just a few days data. But, of course, Loop is sending treatments, etc, all the time.

![image](https://cloud.githubusercontent.com/assets/2892/24934933/96985d16-1ed3-11e7-8498-b1632b29849e.png)

PieterGit commented:
I noticed the light weight monitoring tool Monitorix has a MongoDB monitoring option on their master branch (see https://github.com/mikaku/Monitorix/issues/38 
I just upgraded my Nightscout from version 0.9.3-dev-20170318 head 4e7d5e7, to 639e52b and I'll monitor the memory usage. I reboot my local Nightscout about once a week to keep it usable. I still use node 4.5.0 and npm 2.15.9.

shanselman commented:
OK so here's an update on my site over the week. So what are we thinking this is? The treatments sent by loop and NS spinning over them?

![image](https://cloud.githubusercontent.com/assets/2892/25051240/7c5ab08c-2100-11e7-87d4-6fffd433f754.png)

ELUTE commented:
I know that having COB enabled will cause increased errors... not sure if that would affect CPU

shanselman commented:
@ELUTE Good info. I can try turning that off temporarily.

sulkaharo commented:
Friend who has experience with Node development just said the 7.x series fixes at least half of the closure leaks in Node (compared to 6.1); we're probably bumping our head on Node bugs and should try what the behavior is under 7.9.

jasoncalabrese commented:
@ELUTE I've notice that too with COB, I think it's related to getting cob from devicestatus for openaps/Loop, I opened a new issue for that and have an idea where it is https://github.com/nightscout/cgm-remote-monitor/issues/2550

@shanselman @sulkaharo I'm thinking that we could merge this, it's moving in the right direction, the memory issue weren't solved, but I don't think they were made worse either. It should be easy to test with node 7.9 locally and see if that improves the leak with system objects.

jasoncalabrese commented:
Just fixed a the clipping bug on iOS when rotating, since this is still open I added it here.  Makes a big difference for me when rotating now that I have an iPhone..

jasoncalabrese commented:
Since no one is objecting, I'm going to merge

francescaneri commented:
I found slowdowns in uploading dexcom g5 data. confirmed?
